homeassistant/components/bond/fan.py

```diff
@@ -199,10 +199,6 @@ async def async_turn_on(
 
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set the preset mode of the fan."""
-        if preset_mode != PRESET_MODE_BREEZE or not self._device.has_action(
```

Comment:
Do we need to keep `self._device.has_action(Action.BREEZE_ON)`?

Reply:
Why, it is already checked upon. But may be it does not work if  it is dynamical.

Reply:
Moved back the has_action check.